app/models/user.rb

```diff
@@ -1,19 +1,21 @@
 class User < ApplicationRecord
   has_secure_password
 
-  before_save :downcase_nickname
+  before_validation :downcase_nickname
 
   validates :email,
-    presence: true,
-    uniqueness: true,
+    db_presence: true,
+    db_uniqueness: true,
     format: { with: /\A[a-z\d_+.\-]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i }
```

Comment:
Don't limit the email pattern so much. You can't even imagine what real email can exist. Please look [valid email regexp](https://emailregex.com/#crayon-5dcf0d9dc15ec916764848)

By regexp limiting, you create inconvenience to your users

Therefore, I believe that the main criterion is the presence of `@`. If this is valid email, then the user will receive email. If not, then will not. It's quite simple :)